### PR TITLE
Board Class Part 1 

### PR DESCRIPTION
**Pull request by willfox0409**
Board Class

- initialized with @cells Hash that contains the cell objects/coordinates that make up the board 
- wrote #valid_coordinate?, #place methods
- implemented helper methods for parent #valid_placement? method 
         > > empty_cells?, valid_length? horizontal?, vertical?
- combined all helper methods within #valid_placement? method 
- stopped at the board rendering portion! 

Board Tests
- wrote tests for all methods, all tests passing 
- could use some more edge cases and perhaps new titles for tests? Let's tackle that if we have time to refactor !

### CODE REVIEW

**Comment by willfox0409** [comment]
Hey @Coralruschak ! I'm happy with how this class is set-up. The only thing I think we could refactor are the tests, down the line if we have time of course, just to cover more edge cases!

File: lib/board.rb [hunk 1]
```diff
@@ -0,0 +1,74 @@
+class Board
+    attr_reader :cells
+    
+    def initialize
```
Comment (Coralruschak) [inline lib/board.rb:4]: I'm wondering if we can use range to shorten this? Definitely a for later thing, but something to consider.

File: lib/board.rb [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+            "D3" => Cell.new("D3"),
+            "D4" => Cell.new("D4")
+        }
+    end
+
+    def valid_coordinate?(coordinate)
+        @cells.has_key?(coordinate)
+    end
+
+    def place(ship, coordinates)
```
Comment (Coralruschak) [inline lib/board.rb:29]: I think we need to update this if an if statement so that a ship is only placed if the placement is valid

Comment (willfox0409) [reply lib/board.rb:29]: Great catch, you'r totally right, the ship can only be placed if it is a 'valid placement'. Go ahead and merge, and we can make that fix on a new branch. Thanks!

File: lib/board.rb (hunk 1 above)
Comment (willfox0409) [reply lib/board.rb:4]: sweet, let's add this to our potential refactors

File: lib/board.rb (hunk 2 above)
Comment (Coralruschak) [reply lib/board.rb:29]: After further discussion, we realized the method was best left as is and that logic was best left to be implemented in the Game class
